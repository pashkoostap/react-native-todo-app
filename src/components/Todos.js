import React from "react";
import { FlatList, RefreshControl } from "react-native";

import Todo from "./Todo";

const Todos = ({ todos = [], deleteTodo }) => (
  <FlatList
    data={todos}
    renderItem={({ item }) => <Todo todo={item} deleteTodo={deleteTodo} />}
    keyExtractor={item => item.id}
  />
);

export default Todos;
