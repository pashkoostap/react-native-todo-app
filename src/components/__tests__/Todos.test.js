import React from "react";
import renderer from "react-test-renderer";

import Todos from "../Todos";

describe("Todos", () => {
  it("should match the snapshot", () => {
    const todos = [
      { title: "todo 1", id: "id_1" },
      { title: "todo 2", id: "id_2" },
      { title: "todo 3", id: "id_3" }
    ];
    const rendered = renderer.create(<Todos todos={todos} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
