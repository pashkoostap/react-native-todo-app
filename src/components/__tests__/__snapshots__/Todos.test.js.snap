// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos should match the snapshot 1`] = `
<RCTScrollView
  data={
    Array [
      "todo 1",
      "todo 2",
      "todo 3",
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  invertStickyHeaders={undefined}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#f2f2f2",
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flex": 1,
            "opacity": 1,
            "padding": 20,
            "width": "100%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#f2f2f2",
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flex": 1,
            "opacity": 1,
            "padding": 20,
            "width": "100%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#f2f2f2",
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "flex": 1,
            "opacity": 1,
            "padding": 20,
            "width": "100%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
